src/oemof/solph/constraints/storage_level.py

```diff
@@ -0,0 +1,177 @@
+):
+    r"""
+    Add constraints to implement storage content based access.
+
+    As a GenericStorage just allows exactly one input and one output,
+    an additional bus, the multiplexer_bus, is used for the connections.
+    Note that all Flow objects connected to the multiplexer_bus have to have
+    a nominal_value.
+
+    Parameters
+    ----------
+    model : oemof.solph.Model
+        Model to which the constraint is added.
+    name : string
+        Name of the multiplexer.
+    storage_component : oemof.solph.components.GenericStorage
+        Storage component whose content should mandate
+        the possible inputs and outputs.
+    multiplexer_bus : oemof.solph.Bus
+        Bus which connects the input and output levels to the storage.
+    input_levels : dictionary with oemof.solph.Bus as keys and float as values
+        Dictionary of buses which act as inputs and corresponding levels
+    output_levels : dictionary with oemof.solph.Bus as keys and float as values
+        Dictionary of buses which act as outputs and corresponding level
+
+    Verbose description can be found in https://arxiv.org/abs/2211.14080
+    """
+
+    def _outputs():
+        OUTPUTS = po.Set(initialize=output_levels.keys())
+        setattr(model, f"{name}_OUTPUTS", OUTPUTS)
+
+        active_output = po.Var(
+            OUTPUTS, model.TIMESTEPS, domain=po.Binary, bounds=(0, 1)
+        )
+        setattr(model, f"{name}_active_output", active_output)
+
+        constraint_name = f"{name}_output_active_constraint"
+
+        def _output_active_rule(m):
+            r"""
+            .. math::
+                y_n \le E(t) / E_n
+            """
+            for t in m.TIMESTEPS:
+                for o in output_levels:
+                    getattr(m, constraint_name).add(
+                        (o, t),
+                        m.GenericStorageBlock.storage_content[
+                            storage_component, t + 1
+                        ]
+                        / storage_component.nominal_storage_capacity
+                        >= active_output[o, t] * output_levels[o],
+                    )
+
+        setattr(
+            model,
+            constraint_name,
+            po.Constraint(
+                OUTPUTS,
+                model.TIMESTEPS,
+                noruleinit=True,
+            ),
+        )
+        setattr(
+            model,
+            constraint_name + "build",
+            po.BuildAction(rule=_output_active_rule),
+        )
+
+        # Define constraints on the output flows
+        def _constraint_output_rule(m, o, p, t):
+            return (
+                m.flow[multiplexer_bus, o, p, t]
+                / m.flows[multiplexer_bus, o].nominal_value
+                <= active_output[o, t]
+            )
+
+        setattr(
+            model,
+            f"{name}_output_constraint",
+            po.Constraint(
+                OUTPUTS,
+                model.PERIODS,
+                model.TIMESTEPS,
```

Comment:
Couldn't you instead write model.TIMEINDEX here? This is what is used fo other flows.